src/bootstrap/src/core/build_steps/test.rs

```diff
@@ -1956,7 +1956,7 @@ NOTE: if you're sure you want to do this, please open an issue as to why. In the
         }
 
         if builder.config.profiler_enabled(target) {
-            cmd.env("RUSTC_PROFILER_SUPPORT", "1");
+            cmd.arg("--profiler-support");
         }
```

Comment:
Looking at the history of this flag, there doesn't seem to be any particular reason for it to be an environment variable instead of a command-line flag.

The main reason I'm changing it over is to make it easier to write unit tests for the ignore-mode changes.